src/array/cuda/rowwise_sampling_prob.cu

```diff
@@ -392,6 +393,74 @@ __global__ void _CSRRowWiseSampleReplaceKernel(
   }
 }
 
+template <typename IdType, typename DType, typename BoolType>
+__global__ void _GenerateFlagsKernel(
+    int64_t n, const IdType* idx, const DType* values, DType criteria, BoolType* output) {
+  int tx = blockIdx.x * blockDim.x + threadIdx.x;
+  const int stride_x = gridDim.x * blockDim.x;
+  while (tx < n) {
+    output[tx] = (values[idx ? idx[tx] : tx] != criteria);
+    tx += stride_x;
+  }
+}
+
+template <DGLDeviceType XPU, typename IdType, typename DType, typename MaskGen>
+COOMatrix COOGeneralRemoveIf(COOMatrix coo, MaskGen maskgen) {
```

Comment:
can we use reference or const ref for these arguments to avoid potential copy overhead?

Reply:
This is a very good point.  We should scan the C++ code and use const references whenever possible.  I'll do this only for the new functions in this PR for now.
@frozenbugs for awareness.